humble/Dockerfile

```diff
@@ -21,9 +21,23 @@ SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 RUN apt-get update && apt-get install -y \
     ca-certificates curl gnupg2 lsb-release
 
-# Set up ROS2 repository
-RUN curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg && \
-    echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/ros-archive-keyring.gpg] http://packages.ros.org/ros2/ubuntu $(. /etc/os-release && echo $UBUNTU_CODENAME) main" | tee /etc/apt/sources.list.d/ros2.list > /dev/null
+# Set Locale before setting up ROS2 repository
+RUN apt update && apt install locales && \
```

Comment:
Nit : Could you use apt-get everywhere and include -y flags with installs to avoid user inputs.